addons/payment_razorpay_oauth/models/payment_provider.py

```diff
@@ -131,6 +131,19 @@ def action_razorpay_create_webhook(self):
             },
         }
 
+    # === CONSTRAINT METHODS === #
+
+    @api.constrains('state')
+    def _check_razorpay_credentials_are_set_before_enabling(self):
+        """ Check that the Razorpay credentials are valid.
+
+        :raise ValidationError: If the Razorpay credentials are not valid.
+        """
+        for provider in self.filtered(lambda p: p.code == 'razorpay' and p.state == 'enabled'):
```

Comment:
Should we maybe do this (didn't test if it makes sense)?

```suggestion
        for provider in self.filtered(lambda p: p.code == 'razorpay' and p.state != 'disabled'):
```